template.tpl

```diff
@@ -978,18 +978,23 @@ const onsuccess = () => {
       const eventProperties = makeTableMap(data.eventProperties || [], 'name', 'value');
 
       // Convert comma-separated groupName into an array of groupNames
-      const eventOptions = makeTableMap((data.trackEventGroups || []).map(group => {
+      const groups = makeTableMap((data.trackEventGroups || []).map(group => {
         return {
           eventGroupType: group.eventGroupType,
           eventGroupName: group.eventGroupName && group.eventGroupName.indexOf(',') > -1 ? stringToArrayAndTrim(group.eventGroupName) : group.eventGroupName
         };
       }), 'eventGroupType', 'eventGroupName') || {};
       
+      const eventOptions = {};
+
       if (data.trackTimestamp) {
         eventOptions.time = normalize(data.trackTimestamp);
       }
       
-      _amplitude(instanceName, 'track', data.eventType, eventProperties, eventOptions);
+      _amplitude(instanceName, 'track', {
+        event_type: data.eventType, 
+        groups: groups
```

Comment:
If `groups` used to be `eventOptions` is the structure correct here? Should it be `...groups` instead?

Reply:
There is no `groups` existing in the [EventOptions](https://github.com/amplitude/Amplitude-TypeScript/blob/main/packages/analytics-types/src/base-event.ts#L12) type. The second parameter is the Event type. 
And the function signature is `track(eventInput: BaseEvent | string, eventProperties?: Record<string, any>, eventOptions?: EventOptions) `.
I didn't get why we need to be `...groups`?

Reply:
I see it matches the other or so maybe it's ok. But the variable used to be called `eventOptions` which makes it seem like it is an `EventOptions` object. Now it is renamed `groups` with the same structure and set as "groups" on the `event object`. Is that as designed? Wouldn't `event.groups` be a different object type with different structure?

Reply:
Hi,

This PR fixes what used to be a confusion between `eventOptions` and `groups` (my mistake for misreading the documentation). Previously, I had bundled groups together with event options into the same object. I now learned that groups need to part of the object together with `event_type`, and eventOptions is a separate separate thing.

So this PR separates `groups` and `eventOptions`, mainly.